From: Rob Clark <rob@ti.com>
Date: Thu, 5 Apr 2012 08:25:37 -0500
Subject: [PATCH] dss: GLOBAL_BUFFER hack
Patch-Mainline: No, real fix still in development

For 2d scanout at higher resolutions we need to increase the gfx
pipe dma buffer.. since we aren't using writeback pipe, assign
it's buffers to gfx.

Signed-off-by: Alexander Graf <agraf@suse.de>

diff --git a/drivers/video/omap2/dss/dispc.c b/drivers/video/omap2/dss/dispc.c
index 4749ac3..7f5f602 100644
--- a/drivers/video/omap2/dss/dispc.c
+++ b/drivers/video/omap2/dss/dispc.c
@@ -2241,6 +2241,9 @@ int dispc_ovl_enable(enum omap_plane plane, bool enable)
 {
 	DSSDBG("dispc_enable_plane %d, %d\n", plane, enable);
 
+	// XXX quick hack.. give WB buffers to gfx:
+	dispc_write_reg(DISPC_GLOBAL_BUFFER, 0x006D2240);
+
 	REG_FLD_MOD(DISPC_OVL_ATTRIBUTES(plane), enable ? 1 : 0, 0, 0);
 
 	return 0;
diff --git a/drivers/video/omap2/dss/dispc.h b/drivers/video/omap2/dss/dispc.h
index f278080..318b763 100644
--- a/drivers/video/omap2/dss/dispc.h
+++ b/drivers/video/omap2/dss/dispc.h
@@ -35,6 +35,7 @@
 #define DISPC_GLOBAL_ALPHA		0x0074
 #define DISPC_CONTROL2			0x0238
 #define DISPC_CONFIG2			0x0620
+#define DISPC_GLOBAL_BUFFER		0x0800
 #define DISPC_DIVISOR			0x0804
 
 /* DISPC overlay registers */
